app/src/components/common/SharingModal/index.tsx

```diff
@@ -1,51 +1,74 @@
 import React, { useMemo } from "react";
 import { RQModal } from "lib/design-system/components";
 import { Tabs } from "antd";
+import { ShareLinkView } from "./ShareLinkView";
 import { HiOutlineShare } from "@react-icons/all-files/hi/HiOutlineShare";
+import { PiWarningCircleBold } from "@react-icons/all-files/pi/PiWarningCircleBold";
+
 import type { TabsProps } from "antd";
 import { SharingOptions } from "./types";
 import "./index.css";
 
-export const SharingModal: React.FC = () => {
+interface ModalProps {
+  isOpen: boolean;
+  toggleModal: () => void;
+  rulesToShare: string[];
+}
+
+export const SharingModal: React.FC<ModalProps> = ({ isOpen, toggleModal, rulesToShare = null }) => {
   const sharingOptions: TabsProps["items"] = useMemo(
     () => [
-      {
-        key: SharingOptions.WORKSPACE,
-        label: "Share in workspace",
-        children: "SHARE WITHIN WORKSPACE HERE",
-      },
+      // {
+      //   key: SharingOptions.WORKSPACE,
+      //   label: "Share in workspace",
+      //   children: "SHARE WITHIN WORKSPACE HERE",
+      // },
       {
         key: SharingOptions.SHARE_LINK,
         label: "Shared list",
-        children: "SHARE SHARED LIST HERE",
+        children: <ShareLinkView rulesToShare={rulesToShare} />,
       },
       {
         key: SharingOptions.DOWNLOAD,
         label: "Download",
         children: "DOWNLOAD FROM HERE",
       },
     ],
-    []
+    [rulesToShare]
   );
 
+  console.log({ rulesToShare });
   const handleSharingOptionsChange = (key: SharingOptions) => {
     console.log({ key });
     //TODO: track share option tab clicked here
   };
 
   return (
-    <RQModal wrapClassName="sharing-modal-wrapper" title="Share rule" open={true} closeIcon={<></>} centered>
+    <RQModal
+      wrapClassName="sharing-modal-wrapper"
+      title="Share rule"
+      open={isOpen}
+      destroyOnClose
+      onCancel={toggleModal}
+      centered
+    >
       <div className="rq-modal-content">
         <div className="sharing-modal-header">
           <HiOutlineShare /> Share rule
         </div>
-        <Tabs
-          defaultActiveKey={SharingOptions.WORKSPACE}
-          items={sharingOptions}
-          onChange={handleSharingOptionsChange}
-        />
+        {rulesToShare?.length ? (
+          <Tabs
+            defaultActiveKey={SharingOptions.SHARE_LINK}
+            items={sharingOptions}
+            onChange={handleSharingOptionsChange}
+          />
+        ) : (
+          <div className="sharing-modal-empty-view sharing-modal-body">
+            <PiWarningCircleBold />
+            <div className="title text-white text-bold">Please select the rules that you want to share</div>
+          </div>
```

Comment:
Create a separate component for empty state, so that it can be use in other tabs also